lib/Serialization/Serialization.cpp

```diff
@@ -1819,9 +1835,18 @@ void Serializer::writeCrossReference(const Decl *D) {
   auto val = cast<ValueDecl>(D);
   auto ty = val->getInterfaceType()->getCanonicalType();
   abbrCode = DeclTypeAbbrCodes[XRefValuePathPieceLayout::Code];
+  IdentifierID iid;
+  switch (val->getBaseName().getKind()) {
+  case DeclBaseName::Kind::Normal:
+    iid = addIdentifierRef(val->getBaseName().getIdentifier());
```

Comment:
I'm fine with `addDeclBaseNameRef`. Just because the "reference" is a hardcoded value for known names doesn't change the general purpose.